examples/libadwaita/leaflet_sidebar.rs

```diff
@@ -0,0 +1,135 @@
+                        }
+                    },
+
+                    gtk::ListBox {
+                        set_selection_mode: gtk::SelectionMode::Single,
+                        add_css_class: "navigation-sidebar",
+
+                        adw::ActionRow {
+                            set_title: "Section 1",
+                        },
+
+                        adw::ActionRow {
+                            set_title: "Section 2",
+                        },
+
+                        adw::ActionRow {
+                            set_title: "Section 3",
+                        },
+
+                        connect_row_selected[sender] => move |_, row| {
+                            if let Some(row) = row {
+                                sender.input((row.index() + 1) as u32);
+                            }
+                        }
+                    }
+                },
+
+                gtk::Box {
+                    set_orientation: gtk::Orientation::Vertical,
+                    set_hexpand: true,
+
+                    #[name = "content_header"]
+                    adw::HeaderBar {
+                        #[name = "back_button"]
+                        pack_start = &gtk::Button {
+                            set_icon_name: "go-previous-symbolic",
+                            connect_clicked[leaflet] => move |_| {
+                                leaflet.navigate(adw::NavigationDirection::Back);
+                            }
+                        },
+
+                        #[wrap(Some)]
+                        set_title_widget = &adw::WindowTitle {
+                            set_title: "Content",
+                        }
+                    },
+
+                    gtk::Label {
+                        add_css_class: "title-1",
+                        set_vexpand: true,
+
+                        #[watch]
+                        set_text: &format!("Page {}", model.current_section),
+                    }
+                },
+            }
+        }
+    }
+
+    fn update(&mut self, msg: u32, _: ComponentSender<Self>) {
+        self.current_section = msg;
+    }
+
+    fn init(_: (), root: &Self::Root, sender: ComponentSender<Self>) -> ComponentParts<Self> {
+        let model = App { current_section: 1 };
+
+        let widgets = view_output!();
+
+        widgets
+            .leaflet
+            .bind_property("folded", &widgets.sidebar_header, "show-end-title-buttons")
+            .flags(glib::BindingFlags::SYNC_CREATE)
+            .build();
+        widgets
+            .leaflet
+            .bind_property(
+                "folded",
+                &widgets.content_header,
+                "show-start-title-buttons",
+            )
+            .flags(glib::BindingFlags::SYNC_CREATE)
+            .build();
+        widgets
+            .leaflet
+            .bind_property("folded", &widgets.back_button, "visible")
+            .flags(glib::BindingFlags::SYNC_CREATE)
+            .build();
```

Comment:
Technically, this could be expressed within the `view` macro now, but probably it's better to keep it here to make it more readable.

Reply:
Agreed. I had trouble getting `chain` to work with `flags`.